### DISCUSSION TITLE
Why Can't Rows without `subRows` be "Expandable"

### DISCUSSION THREAD

**Discussion by gavinvaske** [TanStack/table#5243] [Ideas]
⚠️ My knowledge of this library is minimum, beware of dumb questions ⚠️ 

# Problem Statement

I am rendering a table of `People`, i.e:

```
const people = [
  {
    name: 'foobar',
    detailsOnlyShownDuringRowExpansion: { ... }
  }
]

useReactTable{
  data: people
}
```

I want the rows of this table to include _some_ details about each person, i.e their name.

However, if a user clicks on a row, I want to utilize the built-in methods from this library that handle row expansion and/or subComponents.

However, [this code in the library](https://github.com/TanStack/table/blob/aec4d3c409f739bcf5f78bf74236ae5852860804/packages/table-core/src/features/Expanding.ts#L326C53-L326C74) only allows a row to be considered "expandable" if the data you gave in `useReactTable` includes a data attribute called `subRows` whose length is greater than 0.

I can create a hack solution using to make the row "expandable"

```
const people = [
  {
    name: 'foobar',
    detailsOnlyShownDuringRowExpansion: { ... },
    subRows: ['useless string that makes this row now considered "expandable"']
  }
]

useReactTable{
  data: people
}
```

But why do I need to? I don't want to have to populate this `subRows` attribute purely to make the library work, there must be a better solution.

**Comment by TonyVeigel** [comment]
https://tanstack.com/table/v8/docs/api/features/expanding#getrowcanexpand
